src/elf/types.rs

```diff
@@ -108,7 +109,7 @@ impl Elf64_Sym {
     #[inline]
     pub fn matches(&self, type_: SymType) -> bool {
         let elf_ty = self.type_();
-        let matches_func = || elf_ty == STT_FUNC;
+        let matches_func = || elf_ty == STT_FUNC || elf_ty == STT_GNU_IFUNC;
```

Comment:
so confusing written like this :) do you think having matches_func as a closure is a perf benefit vs just calculating "is_func" and "is_var" bools unconditionally?

Reply:
Yeah, I didn't even internalize that closures and logical ORs use the same syntax until I saw that. Not sure if there is any performance benefit, it just somehow seemed logical to me. Will change it.